### PR TITLE
Update every ASR config to new Hydra format

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Update every ASR config to new hydra 1.x format

### CODE REVIEW

File: examples/asr/conf/jasper_10x5dr.yaml [hunk 1]
```diff
@@ -159,7 +159,7 @@ model:
 
   optim:
     name: novograd
-    # cls: nemo.core.optim.optimizers.Novograd
+    # _target_: nemo.core.optim.optimizers.Novograd
```
Comment (tkornuta-nvidia) [inline examples/asr/conf/jasper_10x5dr.yaml:162]: ehehe, even in comments... respect! ;)

**Review by tkornuta-nvidia — APPROVED** [review]

File: nemo/core/classes/modelPT.py [hunk 2]
```diff
@@ -98,23 +98,24 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
...
             raise ValueError(
                 f"trainer constructor argument must be either None or pytroch_lightning.Trainer. But got {type(trainer)} instead."
             )
         super().__init__()
+
+        # Convert config to a DictConfig
+        cfg = model_utils.convert_model_config_to_dict_config(cfg)
+
+        # Convert config to support Hydra 1.0+ instantiation
+        cfg = model_utils.maybe_update_config_version(cfg)
```
Comment (tkornuta-nvidia) [inline nemo/core/classes/modelPT.py:111]: I love the new name ;)

**Comment by titu1994** [comment]
Heh let's hope this is the last time we ever need such a heavy update to configs

**Comment by tkornuta-nvidia** [comment]
🤞  still, I expect one more change related to this project:
https://github.com/pytorch/hydra-torch
once it will be mature enough...

**Comment by romesco** [comment]
@tkornuta-nvidia I miss you!

My goal is to have the initial versioned release by the end of January 😉 .

**Comment by tkornuta-nvidia** [comment]
> @tkornuta-nvidia I miss you!
> 
> My goal is to have the initial versioned release by the end of January 😉 .

Hi @romesco sorry for being absent, I was involved in a project that consumed 120.5% of my bandwidth. :(

**Pull request merged**